core/src/main/scala/io/projectglow/sql/expressions/MeanSubstitute.scala

```diff
@@ -47,42 +47,59 @@ case class MeanSubstitute(array: Expression, missingValue: Expression)
   // A value is considered missing if it is NaN, null or equal to the missing value parameter
   def isMissing(lv: NamedLambdaVariable): Predicate = IsNaN(lv) || IsNull(lv) || lv === missingValue
 
-  def getMean: Expression = {
-    val nLv =
-      NamedLambdaVariable("numArg", array.dataType.asInstanceOf[ArrayType].elementType, true)
-    val sLv = NamedLambdaVariable("structArg", StructType.fromDDL("sum double, count long"), true)
+  def getNamedStruct(sumValue: Expression, countValue: Expression): Expression = {
     val sumName = Literal(UTF8String.fromString("sum"), StringType)
     val countName = Literal(UTF8String.fromString("count"), StringType)
-    val getSum = GetStructField(sLv, 0, Some("sum"))
-    val getCount = GetStructField(sLv, 1, Some("count"))
+    namedStruct(sumName, sumValue, countName, countValue)
+  }
 
-    // Sum and count of non-missing values
-    val zeros = namedStruct(sumName, Literal(0d), countName, Literal(0L))
-    // Update sum and count with array value
-    val updateFn = LambdaFunction(
+  def getNumericLambdaVariable(): NamedLambdaVariable =
+    NamedLambdaVariable("numArg", array.dataType.asInstanceOf[ArrayType].elementType, true)
+  def getStructLambdaVariable(): NamedLambdaVariable =
+    NamedLambdaVariable("structArg", StructType.fromDDL("sum double, count long"), true)
+
+  def getSum(lv: NamedLambdaVariable): GetStructField = GetStructField(lv, 0, Some("sum"))
```

Comment:
Think it would be more clear to call the argument `stateStruct` or something like that.